src/catch2/catch_test_macros.hpp

```diff
@@ -142,6 +144,7 @@
   #define TEST_CASE( ... ) INTERNAL_CATCH_TESTCASE( __VA_ARGS__ )
   #define TEST_CASE_METHOD( className, ... ) INTERNAL_CATCH_TEST_CASE_METHOD( className, __VA_ARGS__ )
   #define METHOD_AS_TEST_CASE( method, ... ) INTERNAL_CATCH_METHOD_AS_TEST_CASE( method, __VA_ARGS__ )
+  #define TEST_CASE_FIXTURE( className, ... ) INTERNAL_CATCH_TEST_CASE_FIXTURE( className, __VA_ARGS__ )
```

Comment:
I am not tied to this name in any way. I think it is an OK name because this is the typical behaviour of a fixture in other testing frameworks i.e. the fixture objects is the same fixture object for all test cases that use it. 

I toyed with TEST_CASE_PERSISTENT_FIXTURE and TEST_CASE_WITH_SETUP among other names but they seemed to be a bit too long... In the end, I decided to just go with this name for now and if anyone has a better suggestion, I am more than happy to go with that.

Reply:
> because this is the typical behaviour of a fixture in other testing frameworks i.e. the fixture objects is the same fixture object for all test cases that use it.

This is not true. The most used C++ test framework is GTest, which creates separate fixture instance per test. Catch2's current support for this does the same, and thus also Doctest and other clones. If I remember CppUnit correctly, it also creates a separate instance of fixture per test.

Now you might argue that none of these is a good match, because only Catch2 and its derivatives support defining multiple partial test cases in single test case body, and for GTest and CppUnit we are talking about different test case bodies, but I think of Catch2's test case trees as a short hand for defining multiple test cases, which would make them equivalent.

I am not saying that `TEST_CASE_FIXTURE` is the wrong name, just disputing the assertion about the typical behaviour of a (C++) test framework.

----------------

If I am reading the xUnit docs correctly, it also creates a separate fixture instance per test. Same for Python's unittest.

----------------

Personally I slightly prefer `TEST_CASE_PERSISTENT_FIXTURE`.

Reply:
TEST_CASE_PERSISTENT_FIXTURE it is then!